### PR TITLE
Update mpRestClient-3.0 to tolerate MP 6 and EE 10

### PR DESCRIPTION
**Pull request by Azquelt**
mpRestClient-3.0 will be included in both MP 5 and MP 6.

For #21968

### CODE REVIEW

**Comment by Azquelt** [comment]
I'm currently seeing SecurityManager warnings in SseTest:

```
java.security.AccessControlException: Access denied ("java.lang.RuntimePermission" "modifyThread")java.base/java.security.AccessController.throwACE(AccessController.java:176)
java.base/java.security.AccessController.checkPermissionHelper(AccessController.java:238)
java.base/java.security.AccessController.checkPermission(AccessController.java:385)
java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:322)
com.ibm.ws.kernel.launch.internal.MissingDoPrivDetectionSecurityManager.checkPermission(MissingDoPrivDetectionSecurityManager.java:45)
java.base/java.util.concurrent.ThreadPoolExecutor.checkShutdownAccess(ThreadPoolExecutor.java:748)
java.base/java.util.concurrent.ThreadPoolExecutor.shutdown(ThreadPoolExecutor.java:1373)
java.base/java.util.concurrent.ScheduledThreadPoolExecutor.shutdown(ScheduledThreadPoolExecutor.java:842)
org.jboss.resteasy.concurrent.ContextualExecutorService.shutdown(ContextualExecutorService.java:76)
org.jboss.resteasy.plugins.server.servlet.Servlet3AsyncHttpRequest$Servlet3ExecutionContext$Servlet3AsynchronousResponse.close(Servlet3AsyncHttpRequest.java:322)
org.jboss.resteasy.plugins.server.servlet.Servlet3AsyncHttpRequest$Servlet3ExecutionContext$Servlet3AsynchronousResponse.complete(Servlet3AsyncHttpRequest.java:116)
org.jboss.resteasy.plugins.providers.sse.SseEventOutputImpl.close(SseEventOutputImpl.java:471)
org.jboss.resteasy.plugins.providers.sse.SseEventOutputImpl.close(SseEventOutputImpl.java:116)
mpRestClient20.sse.SseResource.lambda$sendThreeStrings$0(SseResource.java:47)
mpRestClient20.sse.SseResource$$Lambda$451/0x000000002802ba00.run(Unknown Source)
java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
java.base/java.lang.Thread.run(Thread.java:866)

[17/08/2022, 15:01:47:103 BST] 00000051 kernel.launch.internal.MissingDoPrivDetectionSecurityManager I CWWKE0919I: Offending class is: 

     mpRestClient20.sse.SseResource  in  {file:/home/andrew/git/open-liberty/dev/build.image/wlp/usr/servers/mpRestClient20.sse/dropins/sseApp.war}
```

I don't see these warnings being emitted on the MP 4.0 and MP 5.0 repeats.

**Comment by Azquelt** [comment]
I've disabled the `SseTest` for `MP60` for now, and I've got the TCK running as well.

**Comment by Azquelt** [comment]
#libby #build

File: dev/com.ibm.ws.microprofile.rest.client_fat/fat/src/com/ibm/ws/microprofile/rest/client/fat/BasicCdiTest.java
Comment (jhanders34) [inline dev/com.ibm.ws.microprofile.rest.client_fat/fat/src/com/ibm/ws/microprofile/rest/client/fat/BasicCdiTest.java:80]:
```suggestion
        if (JakartaEE9Action.isActive()) {
            remoteAppServer.changeFeatures(Arrays.asList("componenttest-2.0", "restfulWS-3.0", "ssl-1.0", "jsonb-2.0"));
        } else if (JakartaEE10Action.isActive()) {
            remoteAppServer.changeFeatures(Arrays.asList("componenttest-2.0", "restfulWS-3.1", "ssl-1.0", "jsonb-3.0"));
```

File: dev/com.ibm.ws.microprofile.rest.client_fat/fat/src/com/ibm/ws/microprofile/rest/client/fat/BasicEJBTest.java
Comment (jhanders34) [inline dev/com.ibm.ws.microprofile.rest.client_fat/fat/src/com/ibm/ws/microprofile/rest/client/fat/BasicEJBTest.java:71]:
```suggestion
        if (JakartaEE9Action.isActive()) {
            remoteAppServer.changeFeatures(Arrays.asList("componenttest-2.0", "restfulWS-3.0", "ssl-1.0", "jsonb-2.0"));
        } else if (JakartaEE10Action.isActive()) {
            remoteAppServer.changeFeatures(Arrays.asList("componenttest-2.0", "restfulWS-3.1", "ssl-1.0", "jsonb-3.0"));
```

File: dev/com.ibm.ws.microprofile.rest.client_fat/fat/src/com/ibm/ws/microprofile/rest/client/fat/BasicTest.java
Comment (jhanders34) [inline dev/com.ibm.ws.microprofile.rest.client_fat/fat/src/com/ibm/ws/microprofile/rest/client/fat/BasicTest.java:71]:
```suggestion
        if (JakartaEE9Action.isActive()) {
            remoteAppServer.changeFeatures(Arrays.asList("componenttest-2.0", "restfulWS-3.0", "ssl-1.0", "jsonb-2.0"));
        } else if (JakartaEE10Action.isActive()) {
            remoteAppServer.changeFeatures(Arrays.asList("componenttest-2.0", "restfulWS-3.1", "ssl-1.0", "jsonb-3.0"));
```

File: dev/com.ibm.ws.microprofile.rest.client_fat/fat/src/com/ibm/ws/microprofile/rest/client/fat/CollectionsTest.java
Comment (jhanders34) [inline dev/com.ibm.ws.microprofile.rest.client_fat/fat/src/com/ibm/ws/microprofile/rest/client/fat/CollectionsTest.java:78]:
```suggestion
        if (JakartaEE9Action.isActive()) {
            remoteAppServer.changeFeatures(Arrays.asList("componenttest-2.0", "restfulWS-3.0", "ssl-1.0", "jsonb-2.0"));
        } else if (JakartaEE10Action.isActive()) {
            remoteAppServer.changeFeatures(Arrays.asList("componenttest-2.0", "restfulWS-3.1", "ssl-1.0", "jsonb-3.0"));
```

File: dev/com.ibm.ws.microprofile.rest.client_fat/fat/src/com/ibm/ws/microprofile/rest/client/fat/HandleResponsesTest.java
Comment (jhanders34) [inline dev/com.ibm.ws.microprofile.rest.client_fat/fat/src/com/ibm/ws/microprofile/rest/client/fat/HandleResponsesTest.java:71]:
```suggestion
        if (JakartaEE9Action.isActive()) {
            remoteAppServer.changeFeatures(Arrays.asList("componenttest-2.0", "restfulWS-3.0", "ssl-1.0", "jsonb-2.0"));
        } else if (JakartaEE10Action.isActive()) {
            remoteAppServer.changeFeatures(Arrays.asList("componenttest-2.0", "restfulWS-3.1", "ssl-1.0", "jsonb-3.0"));
```

File: dev/com.ibm.ws.microprofile.rest.client_fat/publish/servers/mpRestClient10.headerPropagation/server.xml
Comment (jhanders34) [inline dev/com.ibm.ws.microprofile.rest.client_fat/publish/servers/mpRestClient10.headerPropagation/server.xml:7]:
```suggestion
    <feature>servlet-3.1</feature>
```
jaxrs-2.0 is EE7.

File: dev/com.ibm.ws.microprofile.rest.client_fat/publish/servers/mpRestClient11.produceConsume/server.xml
Comment (jhanders34) [inline dev/com.ibm.ws.microprofile.rest.client_fat/publish/servers/mpRestClient11.produceConsume/server.xml:6]:
```suggestion
    <feature>servlet-3.1</feature>
```
Changing to 4.0 will change the test to run EE8.  I assume we want to keep the test running the same that it was.

File: dev/com.ibm.ws.microprofile.rest.client_fat/fat/src/com/ibm/ws/microprofile/rest/client/fat/JsonbContextTest.java
Comment (jhanders34) [inline dev/com.ibm.ws.microprofile.rest.client_fat/fat/src/com/ibm/ws/microprofile/rest/client/fat/JsonbContextTest.java:61]:
```suggestion
        if (JakartaEE9Action.isActive()) {
            remoteAppServer.changeFeatures(Arrays.asList("componenttest-2.0", "restfulWS-3.0", "ssl-1.0", "jsonb-2.0"));
        } else if (JakartaEE10Action.isActive()) {
            remoteAppServer.changeFeatures(Arrays.asList("componenttest-2.0", "restfulWS-3.1", "ssl-1.0", "jsonb-3.0"));
```

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.org.eclipse.microprofile.rest.client-3.0.feature
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.org.eclipse.microprofile.rest.client-3.0.feature:3]: didn't know where to put this comment, but io.openliberty.mpRestClient.internal.cdi-4.0.feature probably should be updated to beta instead of noship?

Comment (jhanders34) [reply dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.org.eclipse.microprofile.rest.client-3.0.feature:3]: Will have to coordinate with my PR though if you make that change.  Either need to rebase on integration after my change and update the test that will now fail, or I need to wait for your PR to merge and then update the test case before I merge.

File: dev/com.ibm.ws.microprofile.rest.client_fat/fat/src/com/ibm/ws/microprofile/rest/client/fat/BasicCdiTest.java
Comment (Azquelt) [reply dev/com.ibm.ws.microprofile.rest.client_fat/fat/src/com/ibm/ws/microprofile/rest/client/fat/BasicCdiTest.java:80]: This doesn't matter because `remoteAppServer` isn't the server under test, it's just the target for `mpRestClient` to make requests against.

Note that `RepeatTests` doesn't update the features for `mpRestClient10.remoteServer`, only for `mpRestClient10.basic.cdi`. We wouldn't need to change the features for this server at all, except that `ShrinkHelper` will automatically transform apps when we're doing an EE9 or EE10 repeat, so the remote server does need to have EE9 compatible features.

File: dev/com.ibm.ws.microprofile.rest.client_fat/publish/servers/mpRestClient10.headerPropagation/server.xml
Comment (Azquelt) [reply dev/com.ibm.ws.microprofile.rest.client_fat/publish/servers/mpRestClient10.headerPropagation/server.xml:7]: This is a mistake, but it will get overwritten by `MicroProfileActions.repeat()`. The test will never run with the exact configuration in the server.xml.

We should still fix it, but this is why it hasn't caused any failures.

File: dev/com.ibm.ws.microprofile.rest.client_fat/publish/servers/mpRestClient11.produceConsume/server.xml
Comment (Azquelt) [reply dev/com.ibm.ws.microprofile.rest.client_fat/publish/servers/mpRestClient11.produceConsume/server.xml:6]: Hmm, that's unfortunate. The test configuration has the first repeat as `MP20`, which would be EE8 features, but it wouldn't have actually run on EE8 because there were no EE features in the server.xml. If we want this test to run on EE7, we should update the first repeat to use `MP14`.

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.org.eclipse.microprofile.rest.client-3.0.feature
Comment (Azquelt) [reply dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.org.eclipse.microprofile.rest.client-3.0.feature:3]: Sounds like it will be easier to do that as a separate change.

**Review by jim-krueger — APPROVED** [review]

**Comment by Azquelt** [comment]
The issues raised in the code review will be addressed in #22161.

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.org.eclipse.microprofile.rest.client-3.0.feature
Comment (jhanders34) [reply dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.org.eclipse.microprofile.rest.client-3.0.feature:3]: I will add it to my PR now that you have merged this PR.

**Pull request merged**
